File: Android/app/src/main/java/com/dreamproject/inwords/core/util/SchedulersFacade.java
package com.dreamproject.inwords.core.util;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.annotations.NonNull;
import io.reactivex.schedulers.Schedulers;

/**
 * Provides various threading schedulers.
 */

public class SchedulersFacade {
    /**
     * IO thread pool scheduler
     */
    @NonNull
    static public Scheduler io() {
        return Schedulers.io();
    }

    /**
     * Computation thread pool scheduler
     */
    @NonNull
    static public Scheduler computation() {
        return Schedulers.computation();
    }

    /**
     * Main Thread scheduler
     */
    @NonNull
    static public Scheduler ui() {
        return AndroidSchedulers.mainThread();
    }
}

File: Android/app/src/main/java/ru/inwords/inwords/core/util/SchedulersFacade.kt
package ru.inwords.inwords.core.util

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * Provides various threading schedulers.
 */
object SchedulersFacade {
    /**
     * IO thread pool scheduler
     */
    fun io(): Scheduler = Schedulers.io()

    /**
     * Computation thread pool scheduler
     */
    fun computation(): Scheduler = Schedulers.computation()

    /**
     * Main Thread scheduler
     */
    fun ui(): Scheduler = AndroidSchedulers.mainThread()

    /**
     * Single Thread scheduler
     */
    fun single(): Scheduler = Schedulers.single()
}
